### PR TITLE
test: acceptance-tests

### PR DESCRIPTION
**Pull request by mohitpubnub**
* update run-tests

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -17,24 +17,38 @@ jobs:
...
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+          features-path: tests/features
+      - name: Run acceptance tests
         run: |
-          cargo test 
-      - name: Cancel workflow runs for commit on error
-        if: failure()
-        uses: ./.github/.release/actions/actions/utils/fast-jobs-failure
+          cargo test --test contract
```
Comment (Xavrax) [inline .github/workflows/run-tests.yml:46]: Shouldn't we have test modules e.g. `publish`, `subscribe`, etc. instead of just `contract`?

**Review by Xavrax — COMMENTED** [review]
Imo it looks good.

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (mohitpubnub) [reply .github/workflows/run-tests.yml:46]: Exploring more on this.
As per My limited understanding how cargo test works - We need one entry point file with same name. And for cucumber settings one file starts execution of all tests. We can filter it out via tags later on. But the entry point/ file.rs is one.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -12,14 +12,14 @@ defaults:
...
+    name: Unit tests
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
-      - name: Run tests
+      - name: Run unit tests
         run: |
-          cargo test --lib transport
+          cargo test --lib
```
Comment (mohitpubnub) [inline .github/workflows/run-tests.yml:22]: @Xavrax   this line works as per our requirement. Isn't it!?

**Review by kleewho — APPROVED** [review]

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (Xavrax) [reply .github/workflows/run-tests.yml:22]: Does it omits the `tests/` directory ;o?

**Review by Xavrax — APPROVED** [review]
LGTM

**Comment by mohitpubnub** [comment]
> Does it omits the tests/ directory ;o? 

Yes It is.  [here](https://github.com/pubnub/rust/actions/runs/4464757542/jobs/7841195096#step:3:363)
In this action it ran this 8 tests only
image

**Pull request merged**